models/events.go

```diff
@@ -1,114 +1,142 @@
 package models
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/onflow/cadence"
 	"github.com/onflow/flow-go-sdk"
-	"github.com/onflow/flow-go/fvm/evm/types"
-	"golang.org/x/exp/slices"
+	"github.com/onflow/flow-go/fvm/evm/events"
 )
 
 // isBlockExecutedEvent checks whether the given event contains block executed data.
 func isBlockExecutedEvent(event cadence.Event) bool {
 	if event.EventType == nil {
 		return false
 	}
-	return strings.Contains(event.EventType.ID(), string(types.EventTypeBlockExecuted))
+	return strings.Contains(event.EventType.ID(), string(events.EventTypeBlockExecuted))
 }
 
 // isTransactionExecutedEvent checks whether the given event contains transaction executed data.
 func isTransactionExecutedEvent(event cadence.Event) bool {
 	if event.EventType == nil {
 		return false
 	}
-	return strings.Contains(event.EventType.ID(), string(types.EventTypeTransactionExecuted))
+	return strings.Contains(event.EventType.ID(), string(events.EventTypeTransactionExecuted))
 }
 
+// CadenceEvents contains Flow emitted events containing one or zero evm block executed event,
+// and multiple or zero evm transaction events.
 type CadenceEvents struct {
-	events flow.BlockEvents
+	events       flow.BlockEvents  // Flow events for a specific flow block
+	block        *Block            // EVM block (at most one per Flow block)
```

Comment:
```suggestion
	block        *Block            // EVM block (exactly one per Flow block)
```
I believe we should have exactly one EVM Block per Flow Block. Maybe it is worth it to remove the pointer? To make sure that it is never `nil`.